src/python_testing/TC_DeviceBasicComposition.py

```diff
@@ -338,6 +339,41 @@ class RequiredMandatoryAttribute:
                         success = False
                         continue
 
+        # Validate there are attributes in the global range that are not in the required list
+        allowed_globals = [a.id for a in ATTRIBUTES_TO_CHECK]
+        # also allow event list because it's not disallowed
+        event_list_id = 0xFFFA
+        allowed_globals.append(event_list_id)
+        global_range_min = 0x0000_F000
+        mei_range_min = 0x0001_0000
+        attribute_list_id = 0xFFFB
+        for endpoint_id, endpoint in self.endpoints_tlv.items():
+            for cluster_id, cluster in endpoint.items():
+                globals = [a for a in cluster[attribute_list_id] if a >= global_range_min and a < mei_range_min]
+                unexpected_globals = sorted(list(set(globals) - set(allowed_globals)))
+                for unexpected in unexpected_globals:
+                    location = AttributePathLocation(endpoint_id=endpoint_id, cluster_id=cluster_id, attribute_id=unexpected)
+                    self.record_error(self.get_test_name(), location=location,
+                                      problem=f"Unexpected global attribute {unexpected} in cluster {cluster_id}", spec_location="Global elements")
+                    success = False
+
+        # validate that all the returned attributes in the standard clusters contain only known attribute ids
+        for endpoint_id, endpoint in self.endpoints_tlv.items():
+            for cluster_id, cluster in endpoint.items():
+                if cluster_id not in chip.clusters.ClusterObjects.ALL_ATTRIBUTES:
+                    # Skip clusters that are not part of the standard generated corpus (e.g. MS clusters)
+                    continue
+                standard_attributes = [a for a in cluster[attribute_list_id] if a < global_range_min]
+                allowed_standard_attributes = chip.clusters.ClusterObjects.ALL_ATTRIBUTES[cluster_id]
+                unexpected_standard_attributes = sorted(list(set(standard_attributes) - set(allowed_standard_attributes)))
+                for unexpected in unexpected_standard_attributes:
```

Comment:
Does the test plan have checks for manufacturer-scoped invalid values?  i.e. MEIs where the prefix is not 0 and the suffix is larger than 0x4FFF (so not a non-global attribute) but less than 0xF000 (so not a global attribute)?

If not, it really should, and we should check for it here.

Reply:
no, but I don't think that's specified in the spec either. table 70 says the valid range of suffixes for mei is 0x0000 to 0xfffe

Reply:
@cecille You want to look at table 71, which has different allowed suffix ranges based on type for data model types.

Reply:
Also, table 71 disallows MC global attributes, as far as I can tell....

Reply:
This isn't covered in the current test plan. We can consider adding it in a follow up if you open an issue but I want to be really sure of the requirements before we add this to the cert test.

Reply:
The current test plan states:

```
The list SHALL NOT contain any additional values in the standard or scoped range: (0x0000_0000 - 0x0000_4FFF) and (0x0000_F000 - 0x0000_FFFE)

The list MAY contain values in the Manufacturer Extensible Identifier (MEI) range: (0xXXXX_0000 - 0xXXXX_4FFF), where XXXX is the allowed MEI range (0x0001 - 0xFFF1), these values SHALL be ignored.

The list SHALL NOT contain any values in the Test Vendor or invalid range: (0x0000_5000 - 0x0000_EFFF and 0x0000_FFFF), (0xXXXX_5000 - 0xXXXX_FFFF) and (0xFFF1_0000 - 0xFFFF_FFFF), where XXXX is the allowed MEI range (0x0001 - 0xFFF1)
```

So the test plan does have this ranges listed as not allowed?

Reply:
As far as I understand this test step checks this part of the statement:

**The list SHALL NOT contain any additional values in the standard or scoped range: (0x0000_0000 - 0x0000_4FFF) and (0x0000_F000 - 0x0000_FFFE)**

Reply:
I think there is a mistake in the allowed MEI range stated by the Test plan: (0x0001 - 0xFFF1), as far as I can see it should be 0x0001 - 0xFFF0.

Reply:
Re: MEI list: FFF1 is used for SDK testing, and is present in all-clusters app at least. Breaking on FFF1 means we cannot add test items from SDK

Reply:
The Test Vendor MC is defined as 0xFFF1 - 0xFFF4, but my assumption is that these are not allowed on production devices? 🤔 

Manufacturer Code is 0x0001 - 0xFFF0, so either we allow 0x0001 - 0xFFF4 or 0x0001 - 0xFFF0?

Reply:
> We can consider adding it in a follow up if you open an issue but I want to be really sure of the requirements before we add this to the cert test.

Looks like the test plan says this already.  I am fine with a separate PR to handle this part, @cecille, but I don't think it needs test plan changes.

Reply:
The current test plan doesn't disallow MEI in the gloabl range, though, right? Re: table 71 - are you saying global is disallowed in the MEI range because it's not listed as a permitted source? Is "MC" in that table manufacturer... something? cluster?

This is very confusing....I'm looking right at the spec and still not totally sure what's allowed and not. Or why...why restrict the MEI range?

Reply:
OK, let's address this in a follow up and get this PR in. I'm not technically supposed to be here, and I want to make sure we're all on the same page with MEI before it gets added. IMO the MEI range is less concerning than the standard range, so I'd like to get in what we can before SVE.

issue: https://github.com/project-chip/connectedhomeip/issues/28945

Reply:
The way I interpret https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/data_model/Data-Model-Namespace.adoc#ref_mei_data_model_types is that **Attribute ID (Global)** is only allowed for the Standard range, which is also why the TP only states 0xXXXX_0000 - 0xXXXX_4FFF as the allowed range for MEI range, I guess this is called MC in spec so we might want to align on that.

The latter sentence does includes (0xXXXX_5000 - 0xXXXX_FFFF) as a disallowed section.

Reply:
ok ,what's the final call here, then? Add a check for no values in the MEI prefix range above the standard suffix range? and allow prefix values in the test range?

Reply:
https://github.com/project-chip/connectedhomeip/pull/28926 does the checks for no attributes in (0x0000_5000 - 0x0000_EFFF and 0x0000_FFFF), (0xXXXX_5000 - 0xXXXX_FFFF) and (0xFFF1_0000 - 0xFFFF_FFFF), so once this merges we can move it into this test case?

Reply:
@ReneJosefsen OK to resolve this conversation then, since that test will handle the checks I asked for above?

Reply:
I added it to this PR so I could get this convo resolved and get it in. PTAL

Reply:
Sure, then we need to carve that part out of #28926 when we move it into this script.

Reply:
Ok, this is fine for now, resolving so we can merge